### PR TITLE
Bug 1659970: terraform/exec/plugins/vendor: Bump terraform-provider-aws to v2.2.0

### PR DESCRIPTION
**Pull request by wking**
This picks up [a bunch of upstream improvements][1] including terraform-providers/terraform-provider-aws#7734 to address [rhbz#1659970][3].

Generated with a manual `Gopkg.toml` edit followed by:

```console
$ cd terraform/exec/plugins
$ dep ensure
```

with:

```console
$ dep version
dep:
 version     : v0.5.0-31-g73b3afe
 build date  : 2019-02-08
 git hash    : 73b3afe
 go version  : go1.10.3
 go compiler : gc
 platform    : linux/amd64
 features    : ImportDuringSolve=false
```

[1]: https://github.com/terraform-providers/terraform-provider-aws/blob/v2.2.0/CHANGELOG.md
[3]: https://bugzilla.redhat.com/show_bug.cgi?id=1659970

### CODE REVIEW

File: pkg/terraform/exec/plugins/Gopkg.lock [hunk 1]
```diff
@@ -441,6 +552,55 @@
   revision = "51d6538a90f86fe93ac480b35f37b2be17fef232"
   version = "v2.2.2"
 
+[[projects]]
```
Comment (abhinavdahiya) [inline pkg/terraform/exec/plugins/Gopkg.lock]: Ooo, this does not look right, how is pulling in k8s.io/apimachinery ?

Comment (wking) [reply pkg/terraform/exec/plugins/Gopkg.lock]: I dunno.  I'll regenerate tonight with the debugging cranked up to see.

File: pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_eks_cluster_auth.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/kubernetes-sigs/aws-iam-authenticator/pkg/token"
```
Comment (wking) [inline pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_eks_cluster_auth.go]: @abhinavdahiya, here's where Kubernetes is getting pulled in, so I think [this][1] is not actually a problem.

[1]: https://github.com/openshift/installer/pull/1442/#discussion_r267145595

**Comment by wking** [comment]
I've pushed 6c2558b38 -> d1c17b7f8 adding two `override` stanzas and some `=` prefixes to `Gopkg.toml` to get this to compile.  The AWS Terraform provider has a v2.3 out now, but I'll bump to that in follow-up work after we get this in.

**Comment by wking** [comment]
[Through Terraform][1]:

```
Flaky tests:

[Feature:Builds][Conformance] s2i build with a quota  Building from a template should create an s2i build with a quota and run it [Suite:openshift/conformance/parallel/minimal]

Failing tests:

[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Inline-volume (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Pre-provisioned PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
```

So I think we're good to go.

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1442/pull-ci-openshift-installer-master-e2e-aws/4662

**Comment by abhinavdahiya** [comment]
> [Through Terraform](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1442/pull-ci-openshift-installer-master-e2e-aws/4662):
> 
> ```
> Flaky tests:
> 
> [Feature:Builds][Conformance] s2i build with a quota  Building from a template should create an s2i build with a quota and run it [Suite:openshift/conformance/parallel/minimal]
> 
> Failing tests:
> 
> [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
> [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Inline-volume (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
> [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Pre-provisioned PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
> ```
> 
> So I think we're good to go.

do we need the 420K bump to fix [BZ 1659970](https://bugzilla.redhat.com/show_bug.cgi?id=1659970)(IMO user error)  :cry:

**Comment by wking** [comment]
> do we need the 420K bump to fix BZ 1659970(IMO user error)

Lots of other bugfixes in there too, see [the change-log][1].  For example:

> resource/aws_s3_bucket: Prevent NoSuchBucket errors when putting lifecycle configuration on resource creation (terraform-providers/terraform-provider-aws#7930)

[1]: https://github.com/terraform-providers/terraform-provider-aws/blob/v2.2.0/CHANGELOG.md

**Comment by wking** [comment]
Failing tests are openshift/origin#22412, and may be happening every time.

**Comment by wking** [comment]
[e2e-aws][1]:

```
level=error msg="\t* aws_route_table_association.private_routing.0: timeout while waiting for state to become 'success' (timeout: 5m0s)"
```

/retest

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1442/pull-ci-openshift-installer-master-e2e-aws/4700

**Comment by sdodson** [comment]
`level=fatal msg="failed to initialize the cluster: Working towards 0.0.1-2019-03-27-195700: 99% complete: timed out waiting for the condition"` on the last failure

**Pull request merged**